src/core/raster/qgssinglebandpseudocolorrenderer.cpp

```diff
@@ -37,17 +37,23 @@ QgsSingleBandPseudoColorRenderer::QgsSingleBandPseudoColorRenderer( QgsRasterInt
 }
 
 void QgsSingleBandPseudoColorRenderer::setBand( int bandNo )
+{
+  setInputBand( bandNo );
+}
+
+bool QgsSingleBandPseudoColorRenderer::setInputBand( int band )
 {
   if ( !mInput )
   {
-    mBand = bandNo;
-    return;
+    mBand = band;
+    return true;
   }
-
-  if ( bandNo <= mInput->bandCount() || bandNo > 0 )
+  else if ( band <= mInput->bandCount() || band > 0 )
```

Comment:
suspicious condition... likely
```suggestion
  else if ( band <= mInput->bandCount() && band > 0 )
```